backend node orchestration and Rust API server - Categorizing Transactions

jevinjojo opened the pull request:
## Issue 
- Fixes #200


Node Configs & Startup Scripts

* bitcoind_node/bitcoin.conf and start.sh: blocksonly node with wallet enabled
* cmempoold_node/bitcoin.conf and start.sh: mempool receiver node for transaction indexing

Rust API Server (braidpoold)
* src/api.rs: REST endpoints for transaction submission and status
* src/main.rs: server bootstrap and orchestration logic

jevinjojo committed:
Add Rust backend files via WSL-style paths

jevinjojo committed:
backend - Rust API server and configs

jevinjojo commented:
@mcelrath @Sansh2356 I've opened the PR for the backend orchestration. please take a look and let me know if the file structure looks okay. I forgot to remove the username and password from the config files. I may not be able to respond quickly due to upcoming exams. frontend files still need to be committed and will follow in a separate PR.

jevinjojo committed:
Test file

jevinjojo committed:
backend API tests - API endpoint (31)tests , utility functions (29 tests)

jevinjojo committed:
latest

jevinjojo committed:
port change

priyashuu reviewed (commented):
hey @jevinjojo  , please raise your pr against `dev` branch .

zaidmstrr committed:
fix: stabilize str::from_utf8 by adding std:: prefix (#308)

priyashuu reviewed (commented):
I couldn’t proceed with the review because the PR has merge conflicts.
Please make sure to update your branch with the latest `dev` branch and resolve the conflicts on your side.

krishi-agrawal committed:
mobile ui stats revamp (#287)

Signed-off-by: krishi-agrawal <krishi.agrawal26@gmail.com>
Co-authored-by: Bob McElrath <mcelrath@users.noreply.github.com>

krishi-agrawal committed:
dashboard footer revamp (#290)

footer changes

fix ui

Co-authored-by: Bob McElrath <mcelrath@users.noreply.github.com>

s373nZ committed:
refactor: Remove template retry logic in IPC listener. (#312)

Also Updates IPC listener documentation block for `get_template()` via
review feedback from GitHub Copilot.

krishi-agrawal committed:
Standardised Loader Screens for a Consistent UI (#315)

* unify loader screens

* added mapping

---------

Co-authored-by: Bob McElrath <mcelrath@users.noreply.github.com>

priyashuu commented on CONTRIBUTING.md:
```diff
@@ -1,68 +1,57 @@
-# Welcome to BraidPool contributing guide
+# Contributing to Braidpool
```
I think modifying the CONTRIBUTING file is out of scope for this PR. If changes to it are needed, it would be better to raise a separate PR specifically for that.

priyashuu reviewed (changes requested):
I have reviewed your PR and will review further, but I have a few suggestions:

- Please merge the latest dev branch into your branch.
- Check the Copilot suggestions, I agree with most of them, especially regarding the hardcoded RPC URL and credentials.
- Credentials and ports are inconsistent across files ,please fix this.

For now, I’m marking this PR as a draft, since you had marked it as ready for review. You can update it once you’re ready.

s373nZ committed:
Change IPC listener block size heuristic to transaction count (#320)

* feat: Change IPC listener block size heuristic to transaction count.

* refactor: Replace `80` with const BLOCK_HEADER_BYTES in FinalTemplate.

jevinjojo committed:
Credential configuration

jevinjojo commented:
@priyashuu Hi I've made the requested changes. the main updates include replacing the hardcoded credentials with configuration template and other code improvements.  just only need to add inline comments to certain code blocks

priyashuu commented:
> @priyashuu Hi I've made the requested changes. the main updates include replacing the hardcoded credentials with configuration template and other code improvements. just only need to add inline comments to certain code blocks

Please let us know when this PR is ready for review.

jevinjojo commented:
@priyashuu ready for review

priyashuu commented on bitcoind_node/bitcoin.conf:
```diff
@@ -0,0 +1,25 @@
+regtest=1
```
This file currently includes hard-coded RPC credentials. Even though it’s intended for regtest/local use, committing secrets to the repository is not recommended and can raise security concerns during review.
* Keep the real `bitcoin.conf` ignored via `.gitignore`.

priyashuu replied:
> This file currently includes hard-coded RPC credentials. Even though it’s intended for regtest/local use, committing secrets to the repository is not recommended and can raise security concerns during review.
> 
> * Keep the real `bitcoin.conf` ignored via `.gitignore`.

Is it necessary to include this` bitcoin.conf `file in the repository?

For security and flexibility, it might be better if developers could connect to their own local Bitcoin node instead of relying on a committed config file with hard-coded RPC credentials.

priyashuu commented on cmempoold_node/bitcoin.conf:
```diff
@@ -0,0 +1,18 @@
+regtest=1
```
This file currently includes hard-coded RPC credentials. Even though it’s intended for regtest/local use, committing secrets to the repository is not recommended and can raise security concerns during review.

Keep the real bitcoin.conf ignored via .gitignore.

priyashuu commented on Test.md:
```diff
@@ -0,0 +1,303 @@
+**Prerequisites**
```
I tried testing this PR, but I ran into some issues because the test file is very long (304 lines) and contains a lot of repeated commands, as well as some mistakes.

Could you try simplifying the testing process

jevinjojo committed:
rm hardcoded credentials and update test file

priyashuu commented:
Hi Jevin, are you currently working on this pr?

priyashuu commented:
Thank you for your efforts. I am closing this PR as no recent activity or changes have been made.
